test/dropout.cpp

```diff
@@ -39,6 +39,8 @@
 #define WORKAROUND_MLOPEN_ISSUE_2335 \
     ((HIP_PACKAGE_VERSION_FLAT < 3007000000ULL) && MIOPEN_BACKEND_OPENCL)
 
+static int num_test_case = 0;
```

Comment:
@ce1adon Global non-const variables are evil :)

Reply:
Hmm I can't figure out another way to count the number here. Any suggestion?

Reply:
Hi @atamazov  could you chip in to help please? Actually there are two options:
-  Run this dropout test exactly once. I would recommend moving ```static int num_test_case = 0;``` into scope of ```struct dropout_driver : test_driver```, I think we'll have a single instance of static variable among all translation units.
-  I suspect the following random vector filling has been called many times
   ```
   srand(0);
        for(size_t i = 0; i < in.desc.GetElementSize(); i++)
             reserveSpace[i] =
                 static_cast<unsigned char>(float(GET_RAND()) / float(RAND_MAX) > dropout_rate);
   ```
   I have not done the profiling myself, I would guess would this be a perf bottleneck **especially with multi-threading**? Could we fill it once only, and only shuffle in other instances? Or better yet, use other random generator?

Reply:
Solved with macro